Source/WebCore/css/SelectorChecker.cpp

```diff
@@ -710,8 +710,12 @@ bool SelectorChecker::checkOne(CheckingContext& checkingContext, LocalContext& c
         // :host doesn't combine with anything except pseudo elements.
         bool isHostPseudoClass = selector.match() == CSSSelector::Match::PseudoClass && selector.pseudoClass() == CSSSelector::PseudoClass::Host;
         bool isPseudoElement = selector.match() == CSSSelector::Match::PseudoElement;
-        // We can early return when we know it's neither :host, a compound like :is(:host), a pseudo-element.
-        if (!isHostPseudoClass && !isPseudoElement && !selector.selectorList())
+        // FIXME: We do not support combining :host with :not() functional pseudoclass. Combination with functional pseudoclass has been allowed for the useful :is(:host) ; but combining with :not() doesn't sound useful like :host():not(:not(:host))
```

Comment:
```suggestion
        // FIXME: We do not support combining :host with :not() functional pseudoclass.
```

The second part of the sentence isn't a great justification for not doing things the right way, I would just drop it.